tests/integration/components/sl-button-test.js

```diff
@@ -0,0 +1,206 @@
+import { moduleForComponent, test } from 'ember-qunit';
+import hbs from 'htmlbars-inline-precompile';
+import sinon from 'sinon';
+
+moduleForComponent( 'sl-button', 'Integration | Component | sl button', {
+    integration: true
+});
+
+const mockStreamService = {
+    send() {}
+};
+
+const template = hbs`
+    {{#sl-button}}
+        Default Text
+    {{/sl-button}}
+`;
+
+/**
+ * While it appears that core Ember functionality is being tested this test is
+ * ensuring that the implied contract about which DOM element is rendered is
+ * adhered to.
+ */
+
+test( 'Button tag is rendered', function( assert ) {
+    this.render( template );
+
+    assert.ok(
+        this.$( '>:first-child' ).is( ':button' ),
+        'button tag was rendered'
+    );
+});
+
+test( 'Default action is triggered when element is clicked event when clicked', function( assert ) {
+    assert.expect( 1 );
+
+    const externalAction = () => {
+        assert.ok(
+            true,
+            'External action was called'
+        );
+    };
+
+    this.on( 'externalAction', externalAction );
+
+    this.render( hbs`
+        {{#sl-button action="externalAction"}}
+            Default Text
+        {{/sl-button}}
+    ` );
+
+    this.$( '>:first-child' ).click();
+});
+
+test( 'Button is enabled by default', function( assert ) {
+    this.render( template );
+
+    assert.strictEqual(
+        this.$().is( ':disabled' ),
+        false,
+        'Component is disabled by default'
+    );
+});
+
+test( 'Button is disabled when disabled is set to true', function( assert ) {
+    this.render( hbs`
+        {{#sl-button disabled=true}}
+            Default Text
+        {{/sl-button}}
+    ` );
+
+    assert.strictEqual(
+        this.$( '>:first-child' ).is( ':disabled' ),
+        true,
+        'Component becomes disabled'
+    );
+});
+
+test( 'Expected default classes are applied', function( assert ) {
```

Comment:
@notmessenger I can check for the lack of a class by making sure that there is no `btn-sm` `btn-xs` or `btn-lg` which means it's a medium sized button.